### ISSUE TITLE
store playback position

### ISSUE THREAD

**Issue by stefangweichinger** [memen45/SubMusic#12]
Does playback of music remember where you last stopped? For music this might not be relevant, for listening to podcasts it would be great if I could continue listening where I stopped yesterday. In my experience that is not the case right now. And with podcasts of multiple hours length this is kind of annoying. Try to fast forward 90 minutes on a watch while running :-P 

Do you think/know if this might be related to some "this file is new, because it synced today" or so? Or does the music player on the Garmin watches simply not track "playback position"?

_Originally posted by @stefangweichinger in https://github.com/memen45/SubMusic/issues/9#issuecomment-744487051_

**Comment by stefangweichinger** [comment]
I have to test that myself, but just a note: it would be interesting if the watch remembers position for the files not synced via "SubMusic" but are stored locally.

**Comment by stefangweichinger** [comment]
Quick note: on my watch the playback position in a locally stored podcast was remembered when I started playback today.

**Comment by memen45** [comment]
Thanks for creating a new issue. For the locally stored podcast you mean it was started from the folder 'Podcasts'? If so, that is nice and this could be implemented in SubMusic as well (I checked yesterday and it should be possible). 

The issue I have with it is that it is difficult to distinguish between songs and podcasts. Obviously for songs you would not want to store playback position. Do you have a suggestion on how to determine whether a file should store playback information or not?

**Comment by stefangweichinger** [comment]
Yes, the podcast was stored and played from local folder 'Podcasts'. Maybe it would be enough to store position only for the files stored in a specific folder? I could imagine a "Podcasts" folder in the Nextcloud Music folder ... or a playlist "podcasts" (I have that list already).

Although that is only the beginning: if someday your app has podcast support, I assume it needs the capability to have a tree like:

```
nextcloud/music/
├── podcasts
        ├──  subscribed podcast1
        ├──  subscribed podcast[n]

├── playlist1
└── playlist2
```

;-)

maybe also mimick the structure of Garmin local?

three folders "Audiobooks" "Music" "Podcasts" ?

**Comment by memen45** [comment]
Thanks for your input, I looked at the native music player and came up with the following menu tree:

```
.
└── Library/
    ├── Now Playing
    ├── Play All Songs
    ├── Playlists/
    │   └── Playlist 1/
    │       ├── Play Now
    │       └── Podcast Mode (toggle)
    ├── Podcasts
    ├── Sync Settings/
    │   ├── Sync Now
    │   ├── Playlists
    │   ├── (Podcasts)
    │   ├── Server/
    │   │   ├── Server Info
    │   │   └── Test Server
    │   └── About/
    └── About/
        ├── SubMusic Version
        └── Donate
```
The Podcast Mode option should then enable playback position storage for all the items on its list (or at playlist level, then the playback position will only be stored when the specific item is played on this playlist).
I think I will work on this!

**Comment by stefangweichinger** [comment]
Nice, sounds great ;-)

**Comment by memen45** [comment]
Hi, I am wondering if you could give some more feedback on the behaviour you would like. I have implemented the podcast mode in the latest version (0.1.4-echo). If podcast mode is enabled for the playlist, the positions are stored for the songs that are played. In case the song is completely played, but it is selected as the starting song (from playlist -> songs -> song), the playback state will be reset to 0.
Let me know if it works as you would expect and feel free to comment!

**Comment by memen45** [comment]
Closing this now, as it is implemented and seems to work properly. If you experience any problems, don't hesitate to create a new issue!

**Comment by memen45** [comment]
Just wanted to let you know that actual podcast support is now implemented in the app from v0.1.15 onwards! Access by `More` -> `Browse` -> `Podcasts`.
